client/my-sites/comments/comment-navigation/index.jsx

```diff
@@ -51,6 +51,29 @@ const bulkActions = {
 	all: [ 'approve', 'unapprove', 'spam', 'trash' ],
 };
 
+/**
+ * Formats the comment count to show units in thousands
+ *
+ * @example
+ * countAbbreviation( 999 ) returns 999
+ * countAbbreviation( 1234 ) returns 1.2K
+ * countAbbreviation( 27892 ) returns 28K
+ *
+ * @param {Number} count - raw count
+ * @returns {String} - count string
+ */
+const countAbbreviation = function( count ) {
```

Comment:
we need to support localization for this.

![screen shot 2018-01-28 at 02 24 00](https://user-images.githubusercontent.com/233601/35479317-f23f1f1c-03d2-11e8-9ec8-6e42c39acd7c.png)
![screen shot 2018-01-28 at 02 27 20](https://user-images.githubusercontent.com/233601/35479318-f25cd0ac-03d2-11e8-99bb-523b28c6595e.png)
![screen shot 2018-01-28 at 02 28 59](https://user-images.githubusercontent.com/233601/35479330-485f4688-03d3-11e8-88e4-54e34462bdf4.png)

For example, in Spanish the correct abbreviation is `1,7K`. Maybe this will be a better job for `numberFormat`, by adding a `thousands_abbreviation_translation_key` to our translations, with a `compact` option.

Reply:
This is a great catch.
Italy uses the comma as well, but I'm just so used at seeing both formats that I had a hard time realizing it was incorrect.

Although I've tried to wire up `numberFormat` quickly and dirtily and it looks a bit off:

```es6
// Show 2 sig figs
if ( count < 10000 ) {
	return `${ this.props.numberFormat( count, { decimals: 1, thousandsSep: '' } ) }K`;
}
```

It doesn't really acknowledge that my locale is `it`, and keeps using the period as decimal separator instead of the comma.
Also it disregards `thousandsSep: ''` as falsy, showing the decimal separator anyway (this time it's a period, which is correct for my locale).

Reply:
I was thinking of something more like:
```jsx
<Count count={ count } compact />
```
and then
```js
const { compact } = this.props;
{ numberFormat( count, { compact } ) }
```
The problem is not just the comma. Some countries use `M` and `Mil` instead of `K`. Localization is hard 😛 .

Reply:
The thousands separator should be easy enough to work around, but are we supporting other variants for this round vs plain metric? @drw158 @megsfulton I know India for example uses completely different counting units.

Reply:
Absolutely!

My point was that, currently, `i18n.numberFormat` has got issues messing with what we need.
It apparently has no way to hide the thousands separator, and it doesn't use the correct decimal separator according to the locale.

Anyways, I went to see how Twitter handles this for some random languages, and...

| English | Italian | Spanish |
| --- | --- | --- |
| <img width="336" alt="screen shot 2018-01-29 at 16 46 47" src="https://user-images.githubusercontent.com/2070010/35522477-584b575c-0514-11e8-98dd-df22d201bc90.png"> | <img width="342" alt="screen shot 2018-01-29 at 16 47 32" src="https://user-images.githubusercontent.com/2070010/35522482-5f1b06a4-0514-11e8-8384-9280e9156f42.png"> | <img width="371" alt="screen shot 2018-01-29 at 16 47 51" src="https://user-images.githubusercontent.com/2070010/35522487-6346603e-0514-11e8-8d28-25804fb99948.png"> |

![](https://i.imgur.com/8QmIp.gif)

> Localization is hard

👍

Reply:
I'd say let's ship this with the the period and the K for now, and then work with the i18n folks to add a compact option to `numberFormat`.

Reply:
Hmm, we could also just save some design time and do what twitter does... I'll try giving it another shot after meetings today, and we can loop in the globalization team to check it

Reply:
It's just that I've tried 3 different languages and I've got 3 different formats.
I can't even imagine how many possibilities we could find when starting to support all languages supported by Calypso.

Like off the top of my head, a "compact" number in any given locale can have:
- `.` or `,` as decimal separator
- `,` or `.` or _nothing_ as thousands separator
- `K`, `M`, `mil`, or any other Unicode character, hopefully emoji excluded, as "thousands" abbreviation
- or... be "expanded" instead (as it happens in Italian).

That's not something that we can solve with some design time, isn't it?

Reply:
I was mostly thinking we could match output for our top supported languages. I had to do something similar for [currency formatting](https://github.com/Automattic/wp-calypso/tree/master/client/lib/format-currency). Though we should probably add this option in the i18n module.